contrib/test_jenkins.sh

```diff
@@ -1,277 +1,484 @@
+	$MAKE clean
+	$MAKE
+	$MAKE distclean
+}
+
+#
+# Build without numa support check
+#
+build_disable_numa() {
+	echo "==== Check --disable-numa compilation option ===="
+	../contrib/configure-release --prefix=$ucx_inst --disable-numa
+	$MAKE clean
+	$MAKE
+	$MAKE distclean
+}
+
+#
+# Build a package in release mode
+#
+build_release_pkg() {
+	echo "==== Build release ===="
+	../contrib/configure-release
+	$MAKE clean
+	$MAKE
+	$MAKE distcheck
+
+	# Show UCX info
+	./src/tools/info/ucx_info -f -c -v -y -d -b -p -w -e -uart
+
+	if [ -x /usr/bin/dpkg-buildpackage ]; then
+		echo "==== Build debian package ===="
+		dpkg-buildpackage -us -uc
+	else
+		echo "==== Build RPM ===="
+		../contrib/buildrpm.sh -s -b
+	fi
+
+	$MAKE distclean
+}
+
+#
+# Build with Intel compiler
+#
+build_icc() {
+	echo 1..1 > build_icc.tap
+	if module_load intel/ics
+	then
+		echo "==== Build with Intel compiler ===="
+		../contrib/configure-devel --prefix=$ucx_inst CC=icc CXX=icpc
+		$MAKE clean
+		$MAKE
+		$MAKE distclean
+		module unload intel/ics
+		echo "ok 1 - build successful " >> build_icc.tap
+	else
+		echo "==== Not building with Intel compiler ===="
+		echo "ok 1 - # SKIP because Coverity not installed" >> build_icc.tap
+	fi
+}
+
+#
+# Compile and run UCP hello world example
+#
+run_ucp_hello() {
+	gcc -o ./ucp_hello_world ${ucx_inst}/share/ucx/examples/ucp_hello_world.c \
+		-lucp -lucs -I${ucx_inst}/include -L${ucx_inst}/lib
 
-echo "Build docs"
-make $make_opt docs
-
-echo "Running ucx_info"
-./src/tools/info/ucx_info -v -f
-./src/tools/info/ucx_info -c
-
-echo "Build without IB verbs"
-../contrib/configure-release --without-verbs
-make $make_opt
-
-if [ -n "$JENKINS_RUN_TESTS" ]; then
-    # Set CPU affinity to 2 cores, for performance tests
-    if [ -n "$EXECUTOR_NUMBER" ]; then
-        AFFINITY="taskset -c $(( 2 * EXECUTOR_NUMBER ))","$(( 2 * EXECUTOR_NUMBER + 1))"
-        TIMEOUT="timeout 160m"
-        TIMEOUT_VALGRIND="timeout 200m"
-    else
-        AFFINITY=""
-        TIMEOUT=""
-        TIMEOUT_VALGRIND=""
-    fi
-
-    # Load newer doxygen
-    module load tools/doxygen-1.8.11
-
-    echo "Build gtest"
-    module load hpcx-gcc
-
-    make $make_opt clean
-
-    # todo: check in -devel mode as well
-    ../contrib/configure-release --with-mpi --prefix=$ucx_inst
-    make $make_opt install
-
-    # Prevent out tests from using installed UCX libraries
-    export LD_LIBRARY_PATH=${ucx_inst}/lib:$LD_LIBRARY_PATH
-
-    # check whether installation is valid (it compiles examples at least)
-    make installcheck
-
-    ucx_inst_ptest=$ucx_inst/share/ucx/perftest
-
-    # hack for perftest, no way to override params used in batch
-    # todo: fix in perftest
-    sed -s 's,-n [0-9]*,-n 1000,g' $ucx_inst_ptest/msg_pow2 | sort -R > $ucx_inst_ptest/msg_pow2_short
-    cat $ucx_inst_ptest/test_types | sort -R > $ucx_inst_ptest/test_types_short
+	# hello-world example
+	tcp_port=$((10000 + EXECUTOR_NUMBER))
```

Comment:
this case would break the test. i think it cannot really be handled correctly, except for letting the test application choose a free port. otherwise there could be a race condition.